src/topology/vector_bundle.lean

```diff
@@ -391,3 +513,267 @@ lemma is_open_map_proj : is_open_map Z.proj :=
+  exact (a.continuous_total_space_mk b).cod_restrict (a.mem_trivialization_at_source b),
+end
+
+lemma to_topological_vector_bundle :
+  @topological_vector_bundle R _ F E _ _ _ _ _ _ _ a.total_space_topology _ :=
+{ total_space_mk_inducing := λ b, a.inducing_total_space_mk_of_inducing_comp b
+    (a.total_space_mk_inducing b),
+  locally_trivial := λ b, ⟨a.trivialization_at b, a.mem_base_pretrivialization_at b⟩ }
+
+end topological_vector_prebundle
+
+end
+
+section pullback
+
+/-! ### Pullback of topological vector bundles -/
+
+open topological_space topological_vector_bundle
+
+variables {R F E} {B' : Type*} [topological_space B'] [topological_space (total_space E)]
+
+@[priority 90, nolint unused_arguments]
+instance pullback.total_space.topological_space {f : B' → B} :
+  topological_space (total_space (E ∘ f)) :=
+induced (pullback_total_space_embedding E f) prod.topological_space
+
+lemma inducing_pullback_total_space_embedding {f : B' → B} :
+  inducing (pullback_total_space_embedding E f) := ⟨rfl⟩
+
+lemma continuous_pullback_total_space_embedding {f : B' → B} :
+  continuous (pullback_total_space_embedding E f) :=
+inducing_pullback_total_space_embedding.continuous
+
+variables (R F)
+
+lemma pullback.continuous_total_space_mk {f : B' → B} {x : B'}
+  [∀ x, topological_space (E x)] [topological_vector_bundle R F E] :
+  continuous (total_space_mk (λ y, E (f y)) x) :=
+begin
+  simp only [continuous_iff_le_induced, pullback.total_space.topological_space, induced_compose,
+    pullback_total_space_embedding, prod.topological_space, le_inf_iff, induced_inf, function.comp],
+  have h := (topological_vector_bundle.total_space_mk_inducing R F E (f x)).induced,
+  simp only at h,
+  exact ⟨by {rw induced_const, exact le_top}, by {rw h, exact le_of_eq rfl}⟩,
+end
+
+variables {R F}
+
+lemma pullback.continuous_proj {f : B' → B} :
+  continuous (bundle.proj (E ∘ f)) :=
+begin
+  rw [continuous_iff_le_induced, pullback.total_space.topological_space,
+    pullback_total_space_embedding, prod.topological_space, induced_inf, induced_compose],
+  exact inf_le_left,
+end
+
+lemma pullback.continuous_lift {f : B' → B} :
+  continuous (pullback.lift E f) :=
+begin
+  rw [continuous_iff_le_induced, pullback.total_space.topological_space,
+    pullback_total_space_embedding, prod.topological_space, induced_inf],
+  simp only [induced_compose, inf_le_right],
+end
+
+variables [∀ x, inhabited (E x)]
+
+/-- A vector bundle trivialization can be pulled back to a trivialization on the pullback bundle. -/
+def topological_vector_bundle.trivialization.pullback (e : trivialization R F E) (f : C(B', B)) :
+  trivialization R F (λ y, E (f y)) := --WEEIIRD... composition?
+{ to_fun := λ z, (z.1, (e (total_space_mk E (f z.1) z.2)).2),
+  inv_fun := λ y,
+    if h : f y.1 ∈ e.base_set then
+      total_space_mk _ y.1 (cast (congr_arg E (e.symm_coe_fst' h))
+      (e.to_local_homeomorph.symm (f y.1, y.2)).2)
+    else ↑(default ((E ∘ f) y.1)),
```

Comment:
I think this would be better as:
```suggestion
  inv_fun := λ y,
    total_space_mk _ y.1 $
      if h : f y.1 ∈ e.base_set then
        (cast (congr_arg E (e.symm_coe_fst' h)) (e.to_local_homeomorph.symm (f y.1, y.2)).2)
      else 0,
```

Reply:
Ok thanks a lot!